#! /bin/sh -e

# DP: Fix bashism in makesetup shell script

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        ;;
    *)
	echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
        exit 1
esac
exit 0

--- Modules/makesetup~	2009-11-19 15:14:31.758960233 +0000
+++ Modules/makesetup	2009-11-19 15:14:47.218976182 +0000
@@ -281,7 +281,7 @@
 	-)	;;
 	*)	sedf="@sed.in.$$"
 		trap 'rm -f $sedf' 0 1 2 3
-		echo "1i\\" >$sedf
+		printf "1i\\" >$sedf
 		str="# Generated automatically from $makepre by makesetup."
 		echo "$str" >>$sedf
 		echo "s%_MODOBJS_%$OBJS%" >>$sedf
